juriscraper/pacer/appellate_attachment_page.py

```diff
@@ -0,0 +1,217 @@
+
+    @property
+    def data(self) -> Dict:
+        """Get data back from the query for the matching document entry.
+
+        Appellate attachments is different than the district court.  We have
+        no document_id. Instead we can use the pacer_seq_no or the dls_id
+        to discern which attachment goes with which docket entry.
+
+        Additionally, the main document is not identifiable in attachments.
+        It is presumed to be the document with the lowest pacer_doc_id.
+
+        :return: If lookup fails, an empty dict. Else, a dict containing the
+        following fields:
+            - main_document_id: The id of the main document.
+            - case_id: The pacer case id for this case.
+            - pacer_seq_no - The value of the docket row we came from.
+
+            - attachments: A list of attached items with the following fields:
+                - attachment_number: The attachment number.
+                - description: A description of the item.
+                - page_count: The number of pages for the attachment.
+                - pacer_doc_id: The pacer doc id for the attachment (a str).
+
+        See the JSON objects in the tests for more examples.
+        """
+        if self.is_valid is False:
+            return {}
+
+        # Find the table/tr/td containing a link and return the tr's
+        rows = self.tree.xpath("//tbody/tr/td/a/parent::td/parent::tr")
+        if not rows:
+            logger.info("No documents found on attachment page.")
+            return {}
+
+        result = {
+            "main_document_id": self._get_main_document_id(),
+            "case_id": self._get_pacer_case_id(),
+            "pacer_seq_no": self._get_pacer_seq_no(),
+            "attachments": [],
+        }
+
+        for row in rows:
+            result["attachments"].append(
+                {
+                    "attachment_number": self._get_attachment_number(row),
+                    "description": self._get_description_from_tr(row),
+                    "page_count": self._get_page_count_from_tr(row),
+                    "pacer_doc_id": self._get_pacer_doc_id(row),
+                }
+            )
+        return result
+
+    def _get_main_document_id(self):
+        """Extract the main_document_id.
+
+        The DLS ID  This should be used to identify
+        document number row.
+
+        In appellate attachment page, this is easy to extract with an XPath.
+        :return: main_document_id
+        """
+        try:
+            rows = self.tree.xpath("//tbody/tr/td/a/@href")
+            pre_doc = sorted(row.split("/")[-1] for row in rows)[0]
+            return f"{pre_doc[:3]}0{pre_doc[4:]}"
+        except IndexError:
+            return None
+
+    def _get_attachment_number(self, row) -> int:
+        """ "Return the attachment number for an item.
+
+        :param row: Table row as an lxml element
+        :type: lxml HTML element
```

Comment:
Weird type hints throughout. Time marches on! Can you clean these up, please?

Reply:
Fixed.